src/network/local_socket_reply.cpp

```diff
@@ -252,17 +254,31 @@ void mp::LocalSocketReply::send_request(const QNetworkRequest& request, QIODevic
 
 void mp::LocalSocketReply::read_reply()
 {
-    auto data_ptr = reply_data.data();
+    auto data_ptr = reply_data.data() + reply_offset;
     int bytes_read{0};
 
     do
     {
-        bytes_read = local_socket->read(reply_data.data(), len);
+        if (reply_offset + local_socket->bytesAvailable() > reply_data.length())
+        {
+            reply_data.append(len, '\0');
+            data_ptr = reply_data.data() + reply_offset;
+        }
+
+        bytes_read = local_socket->read(data_ptr, len);
 
+        reply_offset += bytes_read;
         data_ptr += bytes_read;
     } while (bytes_read > 0);
+}
+
+void mp::LocalSocketReply::read_finish()
+{
+    if (local_socket->bytesAvailable())
+        read_reply();
```

Comment:
Any way to write a test to cover this line?  If it's too much work, then I say forget since we have plans for this to go away.

Reply:
No, it's internal to Qt, since we're talking to it over a socket, we've no control of when these signals get fired.

Reply:
Ack